ecommerce/user/utils.py

```diff
@@ -0,0 +1,113 @@
+
+
+async def get_current_user(token: str = Depends(oauth2_scheme), database=Depends(get_db)):
+    credentials_exception = HTTPException(
+        status_code=status.HTTP_401_UNAUTHORIZED,
+        detail="Invalid Credentials",
+        headers={"WWW-Authenticate": "Bearer"},
+    )
+    try:
+        payload = jwt.decode(token, SECRET_KEY, algorithms=[ALGORITHM])
+        email: str = payload.get("sub")
+        if email is None:
+            raise credentials_exception
+    except JWTError:
+        raise credentials_exception
+    user = read_user(email, database)
+    if user is None:
+        raise credentials_exception
+    return user
+
+
+@router.post("/Data/create/", tags=['Data'], status_code=201,
+          response_description="Item have been created successfully")
+async def create_data(first_name, last_name, database=Depends(get_db), user: Users = Depends(get_current_user)):
+    database.add(Data(first_name=first_name, last_name=last_name, user_id=user.id))
+    database.commit()
+
+
+def create_user(username, email, password, database):
+    user = Users(username=username, email=email, password=hash_password(password))
+    database.add(user)
+    database.commit()
+    return UserOut(id=user.id, email=user.email)
+
+
+@router.get(f"/Data/read_one/", tags=['Data'])
+async def read(id, database=Depends(get_db), user: Users = Depends(get_current_user)):
+    statement = select(Data).where(Data.id == id, Data.user_id == user.id).limit(1)
+    data = database.scalar(statement)
+    if data is None:
+        raise HTTPException(status_code=404, detail="Item not found")
+    return data
+
+
+def read_user(email, database):
+    statement = select(Users).where(Users.email == email).limit(1)
+    user = database.scalar(statement)
+    return user
+
+
+@router.get("/Data/read_many/", tags=['Data'])
+async def read_many(limit=None, offset=None, database=Depends(get_db), user: Users = Depends(get_current_user)):
+    statement = select(Data).where(Data.user_id == user.id).order_by(Data.id).limit(limit).offset(offset)
+    data = database.scalars(statement)
+    results = data.all()
+    return results
+
+
+@router.get("/User/read_many/", tags=['User'])
+def read_many_users(limit=None, offset=None, database=Depends(get_db)):
+    statement = select(Users).order_by(Users.id).limit(limit).offset(offset)
+    data = database.scalars(statement)
+    results = data.all()
+    return results
+
+
+@router.put("/Data/update/", tags=['Data'])
+async def update(id, first_name, last_name, database=Depends(get_db), user: Users = Depends(get_current_user)):
+    statement = select(Data).where(Data.id == id, Data.user_id == user.id).limit(1)
+    data = database.scalar(statement)
+    data.first_name = first_name
+    data.last_name = last_name
+    if data is None:
+        raise HTTPException(status_code=404, detail="Item not found")
+    database.commit()
+
+
+@router.delete("/Data/delete/", tags=['Data'])

```

Comment:
```suggestion
@router.delete("/data/{data_id}/", tags=['data'])

```
Same as above. Also, you need to allow user to provide id in the HTTP path, to specify which exact row should be deleted. Not much useful if only first element can be deleted.